x-pack/plugins/infra/server/lib/alerting/metric_threshold/lib/evaluate_rule.ts

```diff
@@ -202,19 +202,35 @@ const getMetric: (
         bucketSelector,
         afterKeyHandler
       )) as Array<Aggregation & { key: Record<string, string>; doc_count: number }>;
-      const groupedResults: Record<string, any> = {};
-      for (const bucket of compositeBuckets) {
-        const key = Object.values(bucket.key).join(', ');
-        const value = getValuesFromAggregations(
-          bucket,
-          aggType,
-          dropPartialBucketsOptions,
-          calculatedTimerange,
-          bucket.doc_count
-        );
-        groupedResults[key] = value;
-      }
-      return groupedResults;
+      // Use unoptimized code from before https://github.com/elastic/kibana/pull/121904/files
```

Comment:
When someone optimizes a rule so that it's no longer CPU bound :tabel-flip:! 

If you can avoid bogging down the CPU altogether that's even better than a worker thread, but I reverted these optimizations to test with.